src/perception/PoseEstimatorModule.cpp

```diff
@@ -90,8 +103,34 @@ bool PoseEstimatorModule::detectObjects(
     Eigen::Isometry3d obj_offset;
     ros::Time t0 = ros::Time(0);
 
-    // get the object name, resource, and offset from database by objectKey
-    mConfigData->getObjectByKey(obj_key, obj_name, obj_resource, obj_offset);
+    // Get the object name, resource, and offset from database by assetKey
+    try
+    {
+      mAssetData->getAssetByKey(asset_key, obj_name, obj_resource, obj_offset);
+    }
+    catch (std::runtime_error& e)
+    {
+      dtwarn << e.what() << std::endl;
+      continue;
+    }
+
+    // Add object to output vector, if available
+    if (detectedObjects)
```

Comment:
I think we need to handle the case when `marker_transform.action` is `DELETE`.

Reply:
Good point. I just push it to the back at the end of the for loop.